mods/win-d-per-monitor.wh.cpp

```diff
@@ -0,0 +1,316 @@
+// ==WindhawkMod==
+// @id              win-d-per-monitor
+// @name            Win+D per monitor
+// @description     Press Win+D to only manage the windows on the monitor where the mouse is located.
+// @description:zh-CN   按下Win+D时 只最小化/还原鼠标所在显示器的窗口
+// @version         1.0.20241214
+// @github          https://github.com/easyatm
+// @include         explorer.exe
+// @compilerOptions -luser32 -lDwmapi
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+# Win+D per monitor
+
+When you press win+d, it only minimizes/restores the windows on the monitor
+where the mouse cursor is located.
+
+按下win+d时,只最小化/还原鼠标所在监视器上的窗口
+
+*/
+// ==/WindhawkModReadme==
+
+#include <dwmapi.h>
+#include <errhandlingapi.h>
+#include <format>
+#include <list>
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+#include <windhawk_api.h>
+#include <windhawk_utils.h>
+#include <windows.h>
+#include <winsvc.h>
+
+static std::wstring ansi_unicode(std::string_view str)
+{
+    if (str.empty())
+        return L"";
+    int size_needed = MultiByteToWideChar(CP_ACP, 0, &str[0], (int)str.size(), NULL, 0);
+    std::wstring wstrTo(size_needed, 0);
+    MultiByteToWideChar(CP_ACP, 0, &str[0], (int)str.size(), &wstrTo[0], size_needed);
+    return wstrTo;
+}
+#define log(...)                                                                                                                                               \
+    {                                                                                                                                                          \
+        std::string gb = std::format(__VA_ARGS__);                                                                                                             \
+        auto ws = ansi_unicode(gb);                                                                                                                            \
+        Wh_Log(L"%s", ws.c_str());                                                                                                                             \
```

Comment:
FYI, you can just use `Wh_Log(L"%S", gb.c_str());` here. Or even better:
```cpp
Wh_Log(L"%S", std::format(__VA_ARGS__).c_str());
```
Then the string will only be constructed if debugging is enabled.